### ISSUE TITLE
Cursor-based pagination does not work at all

### ISSUE THREAD

**Issue by Tenrys** [neo4j/graphql#2437]
**Describe the bug**
Trying to use cursor-based pagination as indicated in the documentation does not work at all, with the Cypher produced not interacting with the GraphQL query.

**Type definitions**
```graphql
type Agent @exclude(operations: [DELETE]) {
  uuid: ID! @id
  archivedAt: DateTime
  
  valuations: [Valuation!]! @relationship(type: "IS_VALUATION_AGENT", direction: OUT)
}
extend type Agent
  @auth(
    rules: [
      { operations: [CREATE], roles: ["Admin"] }
      { where: { archivedAt: null } }
    ]
  )

type Valuation @exclude(operations: [DELETE]) {
  uuid: ID! @id
  archivedAt: DateTime

  agent: Agent! @relationship(type: "IS_VALUATION_AGENT", direction: IN)
}
extend type Valuation
  @auth(
    rules: [
      { where: { archivedAt: null } }
    ]
  )
```

**To Reproduce**

1. Create an agent and then some valuations, over 10.
2. Run the following query with your created UUID
```graphql
query Agents($where: AgentWhere) {
  agents(where: $where) {
    uuid
    valuationsConnection(first: 10) {
      edges {
        node {
          uuid
        }
      }
      pageInfo {
        endCursor
        hasNextPage
      }
    }
  }
}
```
3. You will get all results returned, as indicated by the Cypher printed by the produced debug prints:
```
  @neo4j/graphql:graphql Incoming GraphQL:
Query:
query Agents($where: AgentWhere, $first: Int) {
  agents(where: $where) {
    uuid
    valuationsConnection(first: $first) {
      edges {
        node {
          uuid
        }
      }
      pageInfo {
        endCursor
        hasNextPage
      }
    }
  }
}
Variables:
{
  "where": {
    "uuid": "3ccce208-78b3-445f-aa07-89eba1dc9b4f"
  },
  "first": 10
} +25m
  @neo4j/graphql:execute About to execute Cypher:
Cypher:
MATCH (this:`Agent`)
WHERE (this.uuid = $param0 AND this.archivedAt IS NULL)

CALL {
    WITH this
    MATCH (this)-[this_connection_valuationsConnectionthis0:IS_VALUATION_AGENT]->(this_Valuation:`Valuation`)
    WHERE this_Valuation.archivedAt IS NULL
    WITH { node: { uuid: this_Valuation.uuid } } AS edge
    WITH collect(edge) AS edges
    WITH edges, size(edges) AS totalCount
    RETURN { edges: edges, totalCount: totalCount } AS this_valuationsConnection
}
RETURN this { .uuid, valuationsConnection: this_valuationsConnection } AS this
Params:
{
  "param0": "3ccce208-78b3-445f-aa07-89eba1dc9b4f"
} +25m
  @neo4j/graphql:execute Execute successful, received 1 records +25m
```

**Expected behavior**
I should only be getting 10 items.

**System (please complete the following information):**

* OS: Windows 11 22H2
* Version: @neo4j/graphql@3.12.2
* Node.js version: v16.16.0

**Comment by neo4j-team-graphql** [comment]
Many thanks for raising this bug report @Tenrys. :bug: We will now attempt to reproduce the bug based on the steps you have provided.

Please ensure that you've provided the necessary information for a minimal reproduction, including but not limited to:
* Type definitions
* Resolvers
* Query and/or Mutation (or multiple) needed to reproduce

If you have a support agreement with Neo4j, please link this GitHub issue to a new or existing Zendesk ticket.

Thanks again! :pray:

**Comment by neo4j-team-graphql** [comment]
We've been able to confirm this bug using the steps to reproduce that you provided - many thanks @Tenrys! :pray: We will now prioritise the bug and address it appropriately.

**Comment by neo4j-team-graphql** [comment]
This bug report has been assigned high priority to fix. If you wish to contribute a fix, please branch from `master` and submit your PR with the base set to `master`. Thanks!

**Comment by darrellwarde** [comment]
This should now be fixed in 3.13.0 - please let us know if you're still having issues!
